python/ray/tune/result_grid.py

```diff
@@ -15,30 +15,70 @@
 
 @PublicAPI(stability="beta")
 class ResultGrid:
-    """A set of ``Result`` objects returned from a call to ``tuner.fit()``.
+    """A set of ``Result`` objects for interacting with Ray Tune results.
 
     You can use it to inspect the trials run as well as obtaining the best result.
 
-    The constructor is a private API.
+    The constructor is a private API. This object can only be created as a result of
+    ``Tuner.fit()``.
 
     Usage pattern:
 
     .. code-block:: python
 
+        import random
+        from ray import air, tune
+
+        def random_error_trainable(config):
+            if random.random() < 0.5:
+                return {"loss": 0.0}
+            else:
+                raise ValueError("This is an error")
+
+        tuner = tune.Tuner(
+            random_error_trainable,
+            run_config=air.RunConfig(name="example-experiment"),
+            tune_config=tune.TuneConfig(num_samples=10),
+        )
         result_grid = tuner.fit()
+
         for i in range(len(result_grid)):
             result = result_grid[i]
             if not result.error:
                 print(f"Trial finishes successfully with metric {result.metric}.")
             else:
-                print(f"Trial errors out with {result.error}.")
-        best_result = result_grid.get_best_result()
+                print(f"Trial failed with error {result.error}.")
+
+
+    You can also use `result_grid` for more advanced analysis.
+
+    .. code-block:: python
+
+        # Get the best result based on a particular metric.
+        best_result = result_grid.get_best_result(metric="loss", mode="min")
+
+        # Get the best checkpoint corresponding to the best result.
         best_checkpoint = best_result.checkpoint
-        best_metric = best_result.metric
+
+        # Get a dataframe for the last reported results of all of the trials
+        df = result_grid.get_dataframe()
+
+        # Get a dataframe for the minimum loss seen for each trial
+        df = result_grid.get_dataframe()(metric="loss", mode="min")
 
     Note that trials of all statuses are included in the final result grid.
     If a trial is not in terminated state, its latest result and checkpoint as
     seen by Tune will be provided.
+
+    One may wonder how is ResultGrid different than ExperimentAnalysis. ResultGrid is
+    supposed to succeed ExperimentAnalysis. However, it has not reached the same
+    feature parity yet. For interacting with an existing experiment, located at
+    ``local_dir``, one may do the following:
```

Comment:
Can you make this more active voice?

```suggestion
   ResultGrid will be the successor of the ExperimentAnalysis object
   but is not yet at feature parity. For interacting with an existing experiment, located at
    ``local_dir``, do the following:
```